### PR TITLE
ENCD-5962-avoid-results-from-stale-request

### PR DESCRIPTION
**Pull request by keenangraham**

### CODE REVIEW

File: src/encoded/static/components/top_hits/multi/search.js [hunk 1]
```diff
@@ -13,7 +13,11 @@ import QUERIES from './constants';
 * dropdown using the associated custom results component.
 */
 const Search = ({ children }) => {
+    // User input.
     const [input, setInput] = useState('');
+    // Reference to latest input for use in callback.
+    const inputRef = useRef();
+    inputRef.current = input;
```
Comment (forresttanaka) [inline src/encoded/static/components/top_hits/multi/search.js]: I didn’t think about this before, but I think these two lines can be combined into:
```
const inputRef = useRef(input);
```
I’m about 80% sure that’s equivalent to what you have.

File: src/encoded/static/components/top_hits/multi/search.js [hunk 2]
```diff
@@ -22,6 +26,18 @@ const Search = ({ children }) => {
     // Wait this long after last user input making queries.
     const debounceTime = 200;
 
+    // Comparse searchTerm from query with latest user input.
```
Comment (forresttanaka) [inline src/encoded/static/components/top_hits/multi/search.js]: “Compare”

File: src/encoded/static/components/top_hits/__tests__/top-hits-search-test.js [hunk 3]
```diff
@@ -1,23 +1,369 @@
...
+            },
+        },
+    },
+    num_reduce_phases: 2,
+    timed_out: false,
+    took: 58,
+};
+
+
+// Search for H taks 1000ms to return, search for HBB takes 500ms.
```
Comment (forresttanaka) [inline src/encoded/static/components/top_hits/__tests__/top-hits-search-test.js]: “takes”

**Review by forresttanaka — DISMISSED** [review]
Thanks for working out the complex Jest tests. I don’t think we’ve ever had to work something like that out before. Just a couple feedback items that make (probably) no technical difference.

File: src/encoded/static/components/top_hits/multi/search.js (hunk 1 above)
Comment (keenangraham) [reply src/encoded/static/components/top_hits/multi/search.js]: Hmm, this doesn't seem to work (tests break). I wonder if I should wrap the assignment in useEffect?
```javascript
useEffect(() => {
        inputRef.current = input;
    }, [input]);
```

**Review by forresttanaka — APPROVED** [review]

**Pull request merged**